packages/nimble-components/src/list-option-group/index.ts

```diff
@@ -0,0 +1,121 @@
+import { DesignSystem, FoundationElement } from '@microsoft/fast-foundation';
+import { observable, attr, volatile } from '@microsoft/fast-element';
+import { styles } from './styles';
+import { template } from './template';
+import { ListOption } from '../list-option';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-list-option-group': ListOptionGroup;
+    }
+}
+
+/**
+ * A nimble-styled HTML listbox option
+ */
+export class ListOptionGroup extends FoundationElement {
+    /**
+     * The label for the group.
+     *
+     * @public
+     * @remarks
+     * If a label is also provided via slotted content, the label attribute
+     * will have precedence.
+     */
+    @attr
+    public label?: string;
+
+    /**
+     * The hidden state of the element.
+     *
+     * @public
+     * @defaultValue - false
+     * @remarks
+     * HTML Attribute: hidden
+     */
+    @attr({ mode: 'boolean' })
+    public override hidden = false;
+
+    /**
+     * @internal
+     * This attribute is required to allow use-cases that offer dynamic filtering
+     * (like the Select) to visually hide groups that are filtered out, but still
+     * allow users to use the native 'hidden' attribute without it being affected
+     * by the filtering process.
+     */
+    @attr({ attribute: 'visually-hidden', mode: 'boolean' })
+    public visuallyHidden = false;
+
+    /**
+     * @internal
+     */
+    @attr({ attribute: 'show-top-separator', mode: 'boolean' })
+    public showTopSeparator = false;
+
+    /**
+     * @internal
+     */
+    @attr({ attribute: 'show-bottom-separator', mode: 'boolean' })
+    public showBottomSeparator = false;
+
+    /** @internal */
+    @observable
+    public hasOverflow = false;
+
+    /** @internal */
+    public labelSlot!: HTMLSlotElement;
+
+    /** @internal */
+    @observable
+    public slottedElements: Element[] = [];
+
+    /** @internal */
+    @observable
+    public listOptions: ListOption[] = [];
+
+    /** @internal */
+    @volatile
+    public get labelContent(): string {
+        if (this.label) {
+            return this.label;
+        }
+
+        if (!this.$fastController.isConnected) {
+            return '';
+        }
+
+        const nodes = this.labelSlot.assignedNodes();
+        return nodes
+            .filter(node => node.textContent?.trim() !== '')
+            .map(node => node.textContent?.trim())
+            .join(' ');
+    }
+
+    /**
+     * @internal
+     */
+    public clickHandler(e: MouseEvent): void {
+        e.preventDefault();
+        e.stopImmediatePropagation();
+    }
+
+    protected slottedElementsChanged(): void {
+        this.slottedElements.forEach(e => {
+            if (e instanceof ListOption) {
+                e.slot = 'options-slot';
```

Comment:
can we have the options themselves own configuring their slot?
Also we don't usually have the slot name include the word "slot", maybe just "option" (analogous to `<nimble-tab slot="tab">`)?

Reply:
Can we double check and align the plurality? Looks like we are doing `slot name="options"` here vs the exiting pattern in the tab control which is `slot="tab"`